WalletWasabi/BitcoinCore/Monitoring/P2pReconnector.cs

```diff
@@ -23,12 +23,12 @@ protected override async Task ActionAsync(CancellationToken cancel)
 			try
 			{
 				Logger.LogInfo("Trying to reconnect to P2P...");
-				P2pNode.Disconnect();
+				await P2pNode.DisconnectAsync(cancel).ConfigureAwait(false);
 				await P2pNode.ConnectAsync(cancel).ConfigureAwait(false);
 			}
 			catch
 			{
-				P2pNode.Disconnect();
+				await P2pNode.DisconnectAsync(cancel);
```

Comment:
This cancellation is conceptually different cancellation from the previous ones.

The previous ones cancel the reconnection, this one cancels the disposing of the failed reconnection.

Reply:
If the cancellation kicks in P2pReconnector is signaled to stop whatever it is doing - game over, do not try to do anything. 
Later the node will be properly disposed here:
https://github.com/molnard/WalletWasabi/blob/0c9d83c899a5e2a78d26eb39c7da2e3669b1d293/WalletWasabi.Gui/Global.cs#L730

The concept is fine IMO.

Reply:
How about only calling `DisconnectAsync` if `ConnectAsync` throws an exception and not calling it for the previous `DisconnectAsync` throwing an exception too?

Reply:
I removed this part.